app/controllers/import_controller.rb

```diff
@@ -192,6 +193,24 @@ def validate_bool_field(field, value, row_ind)
     raise ValidationError.new(err_msg, row_ind)
   end
 
+  def validate_race_field(field, value, row_ind)
+    return [] if value.blank?
+    unless value.kind_of?(Array)
+      value = [value]
+    end
+    field.each do |race| 
+      unless VALID_RACES.default.include?(value) || VALID_RACES.exclusive.include?(value)
+        err_msg = "'#{value}' is not an acceptable value for '#{VALIDATION[field][:label]}', acceptable values are any combination of: #{VALID_RACES.default.to_sentence}, or any one of:  #{VALID_RACES.exclusive.to_sentence}"
+        raise ValidationError.new(err_msg, row_ind)
+      end
+      if VALID_RACES.exclusive.include?(value) && field.length > 1
+        err_msg = "'#{value}' is an exclusive value and cannot be combined with other races."
+        raise ValidationError.new(err_msg, row_ind)
+      end
+    end
+    return value
+  end
+
```

Comment:
@holmesie Should we be validating this on enrollment and API as well? Basically should this be converted to a proper validator?

Reply:
That's a good point, @ngfreiter and @tstrass should be involved in this discussion - if at least to strategize.

Reply:
Yeah thanks for pointing this out @Bialogs. This and https://github.com/SaraAlert/SaraAlert/pull/455 will pretty much directly conflict, because this adds a new field, `races`, that won't be validated in #455. If this PR is merged first, I think it is fine to have no validation because no other API fields would yet be getting validated. If #455 is merged first, it of course can't validates `races` since `races` would not exist yet. So I think it makes the most sense to just be aware of this conflict, and whichever PR is merged first, the other can be updated.

Reply:
Checking back since #455 was merged. It looks like this validation was taken out since the `race_unknown`, `race_other` and `race_refused_to_answer` fields are being tracked as booleans now, correct? If that is the case, and the only validation being added in the `ImportController` is validating that the fields are input as booleans, you shouldn't need to add any validation for the API. The API code sets these booleans based on the presence of the `us-core-race` extension, so there isn't a need for validating there.